### PR TITLE
make nspawn backend already working

### PR DESCRIPTION
**Pull request by jscotka**
there were several changes what caused troubles
 * another test image location
 * do not use pull-raw and importing via machinectl, because nobody uses it and causes troubles,
   after dicsussion with systemd devels it is better to do it in own way
 * imrpove tesing of output added chars decode from stdout

verified locally with pytest-3:
```
$ sudo pytest-3 -x tests/integration/test_nspawn_backend.py 
================================================================== test session starts ===================================================================
platform linux -- Python 3.6.6, pytest-3.4.2, py-1.5.4, pluggy-0.6.0 -- /usr/bin/python3
cachedir: .pytest_cache
rootdir: /home/jscotka/git/conu, inifile: pytest.ini
collected 8 items                                                                                                                                        

tests/integration/test_nspawn_backend.py::TestNspawnBackend::test_image PASSED                                                                     [ 12%]
tests/integration/test_nspawn_backend.py::TestNspawnBackend::test_mounts PASSED                                                                    [ 25%]
tests/integration/test_nspawn_backend.py::TestNspawnBackend::test_image_run_foreground PASSED                                                      [ 37%]
tests/integration/test_nspawn_backend.py::TestNspawnBackend::test_backend PASSED                                                                   [ 50%]
tests/integration/test_nspawn_backend.py::TestNspawnBackend::test_image_bootstrapping PASSED                                                       [ 62%]
tests/integration/test_nspawn_backend.py::TestNspawnBackend::test_container_basic PASSED                                                           [ 75%]
tests/integration/test_nspawn_backend.py::TestNspawnBackend::test_container_exit_states PASSED                                                     [ 87%]
tests/integration/test_nspawn_backend.py::TestNspawnBackend::test_volumes PASSED                                                                   [100%]

=============================================================== 8 passed in 84.63 seconds ================================================================
```

### CODE REVIEW

**Comment by jpopelka** [comment]
[test]

**Comment by jpopelka** [comment]
You probably need something like this in requirements.in and requirements.txt
```
backports.lzma ; python_version <= '2.7'
lzma ; python_version >= '3.0'
```

**Comment by jpopelka** [comment]
[test]

**Comment by jscotka** [comment]
> You probably need something like this in requirements.in and requirements.txt
> 
> ```
> backports.lzma ; python_version <= '2.7'
> lzma ; python_version >= '3.0'
> ```

Yeah, that helper, but just partially, next issue is ` ImportError: No module named enum`
means that this is alsno nit in requirements.txt, but this is there for long time and from begining of conu, isn't it @jpopelka

**Comment by TomasTomecek** [comment]
You need to install [enum34 backport](https://stackoverflow.com/a/47731269/909579) on python 2.

PR: https://github.com/user-cont/conu/pull/291

**Comment by jscotka** [comment]
tests passed inside vagrant for nspawn.
just failing port probe test, but this test is not affected by any of changes:
```
default: ++ sudo -E make exec-test
    default: cat pytest.ini
    default: [pytest]
    default: addopts = -vv -m "not release_copr and not release_pypi"
    default: PYTHONPATH=/vagrant pytest-2 
    default: ============================= test session starts ==============================
    default: platform linux2 -- Python 2.7.15, pytest-3.4.2, py-1.5.4, pluggy-0.6.0 -- /usr/bin/python2
    default: cachedir: .pytest_cache
    default: rootdir: /vagrant, inifile: pytest.ini
    default: collecting ... 
    default: collected 129 items
    default: tests/integration/test_backend.py::test_cleanup_containers 
    default: XPASS
    default:          [  0%]
    default: tests/integration/test_backend.py::test_standard_cleanup_tmpdir 
    default: PASSED
    default:    [  1%]
    default: tests/integration/test_backend.py::test_non_cm_backend_tmpdir 
    default: PASSED
    default:      [  2%]
    default: tests/integration/test_backend.py::test_backend_fixture 
    default: PASSED
    default:            [  3%]
    default: tests/integration/test_docker.py::test_parse_reference[registry.fedoraproject.org/fedora:27-result0] 
    default: PASSED
    default:  [  4%]
    default: tests/integration/test_docker.py::test_parse_reference[registry.fedoraproject.org/fedora-result1] 
    default: PASSED
    default:  [  5%]
    default: tests/integration/test_docker.py::test_parse_reference[registry.fedoraproject.org:7890/fedora-result2] 
    default: PASSED
    default:  [  5%]
    default: tests/integration/test_docker.py::test_image 
    default: PASSED
    default:                       [  6%]
    default: tests/integration/test_docker.py::test_image_wrong_types 
    default: PASSED
    default:           [  7%]
    default: tests/integration/test_docker.py::test_container 
    default: PASSED
    default:                   [  8%]
    default: tests/integration/test_docker.py::test_copy_to 
    default: PASSED
    default:                     [  9%]
    default: tests/integration/test_docker.py::test_copy_from 
    default: PASSED
    default:                   [ 10%]
    default: tests/integration/test_docker.py::test_container_create_failed 
    default: PASSED
    default:     [ 11%]
    default: tests/integration/test_docker.py::test_interactive_container 
    default: PASSED
    default:       [ 11%]
    default: tests/integration/test_docker.py::test_container_logs 
    default: PASSED
    default:              [ 12%]
    default: tests/integration/test_docker.py::test_http_client 
    default: PASSED
    default:                 [ 13%]
    default: tests/integration/test_docker.py::test_http_client_context 
    default: PASSED
    default:         [ 14%]
    default: tests/integration/test_docker.py::test_wait_for_status 
    default: PASSED
    default:             [ 15%]
    default: tests/integration/test_docker.py::test_exit_code 
    default: PASSED
    default:                   [ 16%]
    default: tests/integration/test_docker.py::test_blocking_execute 
    default: PASSED
    default:            [ 17%]
    default: tests/integration/test_docker.py::test_nonblocking_execute 
    default: PASSED
    default:         [ 17%]
    default: tests/integration/test_docker.py::test_pull_always 
    default: PASSED
    default:                 [ 18%]
    default: tests/integration/test_docker.py::test_pull_if_not_present 
    default: PASSED
    default:         [ 19%]
    default: tests/integration/test_docker.py::test_pull_never 
    default: PASSED
    default:                  [ 20%]
    default: tests/integration/test_docker.py::test_set_name 
    default: PASSED
    default:                    [ 21%]
    default: tests/integration/test_docker.py::test_run_with_volumes_metadata_check 
    default: PASSED
    default:  [ 22%]
    default: tests/integration/test_docker.py::test_list_containers 
    default: PASSED
    default:             [ 23%]
    default: tests/integration/test_docker.py::test_list_images 
    default: PASSED
    default:                 [ 23%]
    default: tests/integration/test_docker.py::test_build_image 
    default: PASSED
    default:                 [ 24%]
    default: tests/integration/test_docker.py::test_layers 
    default: PASSED
    default:                      [ 25%]
    default: tests/integration/test_docker.py::test_container_metadata 
    default: PASSED
    default:          [ 26%]
    default: tests/integration/test_docker.py::test_image_metadata 
    default: PASSED
    default:              [ 27%]
    default: tests/integration/test_docker.py::test_docker_parameters 
    default: PASSED
    default:           [ 28%]
    default: tests/integration/test_docker.py::test_run_via_api 
    default: PASSED
    default:                 [ 29%]
    default: tests/integration/test_filesystem.py::test_container_read_file 
    default: PASSED
    default:     [ 29%]
    default: tests/integration/test_filesystem.py::test_container_copy_from 
    default: PASSED
    default:     [ 30%]
    default: tests/integration/test_filesystem.py::test_container_get_file 
    default: PASSED
    default:      [ 31%]
    default: tests/integration/test_filesystem.py::test_container_file_is_present 
    default: PASSED
    default:  [ 32%]
    default: tests/integration/test_filesystem.py::test_container_dir_is_present 
    default: PASSED
    default:  [ 33%]
    default: tests/integration/test_filesystem.py::test_container_is_unpacked_well 
    default: PASSED
    default:  [ 34%]
    default: tests/integration/test_filesystem.py::test_image_is_unpacked_well 
    default: PASSED
    default:  [ 35%]
    default: tests/integration/test_filesystem.py::test_image_read_file 
    default: PASSED
    default:         [ 35%]
    default: tests/integration/test_filesystem.py::test_image_copy_from 
    default: PASSED
    default:         [ 36%]
    default: tests/integration/test_filesystem.py::test_image_get_file 
    default: PASSED
    default:          [ 37%]
    default: tests/integration/test_filesystem.py::test_image_file_is_present 
    default: PASSED
    default:   [ 38%]
    default: tests/integration/test_filesystem.py::test_image_dir_is_present 
    default: PASSED
    default:    [ 39%]
    default: tests/integration/test_filesystem.py::test_selinux_context 
    default: PASSED
    default:         [ 40%]
    default: tests/integration/test_helpers.py::test_get_container_output 
    default: PASSED
    default:       [ 41%]
    default: tests/integration/test_k8s.py::TestK8s::test_pod 
    default: SKIPPED
    default:                  [ 41%]
    default: tests/integration/test_k8s.py::TestK8s::test_database_deployment 
    default: SKIPPED
    default:  [ 42%]
    default: tests/integration/test_k8s.py::TestK8s::test_list_pods 
    default: SKIPPED
    default:            [ 43%]
    default: tests/integration/test_k8s.py::TestK8s::test_list_services 
    default: SKIPPED
    default:        [ 44%]
    default: tests/integration/test_k8s.py::TestK8s::test_list_deployments 
    default: SKIPPED
    default:     [ 45%]
    default: tests/integration/test_k8s.py::TestK8s::test_deployment_from_template 
    default: SKIPPED
    default:  [ 46%]
    default: tests/integration/test_k8s.py::TestK8s::test_cleanup 
    default: SKIPPED
    default:              [ 47%]
    default: tests/integration/test_nspawn_backend.py::TestNspawnBackend::test_image 
    default: PASSED
    default:  [ 47%]
    default: tests/integration/test_nspawn_backend.py::TestNspawnBackend::test_mounts 
    default: PASSED
    default:  [ 48%]
    default: tests/integration/test_nspawn_backend.py::TestNspawnBackend::test_image_run_foreground 
    default: PASSED
    default:  [ 49%]
    default: tests/integration/test_nspawn_backend.py::TestNspawnBackend::test_backend 
    default: PASSED
    default:  [ 50%]
    default: tests/integration/test_nspawn_backend.py::TestNspawnBackend::test_image_bootstrapping 
    default: PASSED
    default:  [ 51%]
    default: tests/integration/test_nspawn_backend.py::TestNspawnBackend::test_container_basic 
    default: PASSED
    default:  [ 52%]
    default: tests/integration/test_nspawn_backend.py::TestNspawnBackend::test_container_exit_states 
    default: PASSED
    default:  [ 52%]
    default: tests/integration/test_nspawn_backend.py::TestNspawnBackend::test_volumes 
    default: PASSED
    default:  [ 53%]
    default: tests/integration/test_openshift.py::TestOpenshift::test_oc_s2i_remote 
    default: SKIPPED
    default:  [ 54%]
    default: tests/integration/test_openshift.py::TestOpenshift::test_oc_s2i_local 
    default: SKIPPED
    default:  [ 55%]
    default: tests/integration/test_openshift.py::TestOpenshift::test_oc_s2i_template 
    default: SKIPPED
    default:  [ 56%]
    default: tests/integration/test_s2i.py::test_s2i_extending 
    default: PASSED
    default:                  [ 57%]
    default: tests/integration/test_s2i.py::test_s2i_usage 
    default: PASSED
    default:                      [ 58%]
    default: tests/integration/test_utils.py::test_probes_port 
    default: FAILED
    default:                  [ 58%]
    default: tests/integration/test_utils.py::test_required_binaries_exist 
    default: PASSED
    default:      [ 59%]
    default: tests/integration/test_utils.py::test_command_exists 
    default: PASSED
    default:               [ 60%]
    default: tests/integration/test_utils.py::test_check_docker 
    default: PASSED
    default:                 [ 61%]
    default: tests/integration/test_utils.py::test_run_cmd 
    default: PASSED
    default:                      [ 62%]
    default: tests/unit/test_docker_container.py::test_dr_command_class 
    default: PASSED
    default:         [ 63%]
    default: tests/unit/test_docker_container.py::test_get_port_mappings 
    default: PASSED
    default:        [ 64%]
    default: tests/unit/test_k8s.py::test_port_conversion 
    default: PASSED
    default:                       [ 64%]
    default: tests/unit/test_logging.py::test_capturing 
    default: PASSED
    default:                         [ 65%]
    default: tests/unit/test_logging.py::test_level_setting 
    default: PASSED
    default:                     [ 66%]
    default: tests/unit/test_logging.py::test_logger_has_single_handler 
    default: PASSED
    default:         [ 67%]
    default: tests/unit/test_probe.py::TestProbe::test_in_backgroud 
    default: PASSED
    default:             [ 68%]
    default: tests/unit/test_probe.py::TestProbe::test_exception 
    default: PASSED
    default:                [ 69%]
    default: tests/unit/test_probe.py::TestProbe::test_count 
    default: PASSED
    default:                    [ 70%]
    default: tests/unit/test_probe.py::TestProbe::test_arguments 
    default: PASSED
    default:                [ 70%]
    default: tests/unit/test_probe.py::TestProbe::test_reach_timeout 
    default: PASSED
    default:            [ 71%]
    default: tests/unit/test_probe.py::TestProbe::test_expected_retval 
    default: PASSED
    default:          [ 72%]
    default: tests/unit/test_probe.py::TestProbe::test_concurrency 
    default: PASSED
    default:              [ 73%]
    default: tests/unit/test_utils.py::test_random_str 
    default: PASSED
    default:                          [ 74%]
    default: tests/unit/test_utils.py::test_directory_basic 
    default: PASSED
    default:                     [ 75%]
    default: tests/unit/test_utils.py::test_directory_mode 
    default: PASSED
    default:                      [ 76%]
    default: tests/unit/test_utils.py::test_directory_selinux_bad 
    default: PASSED
    default:               [ 76%]
    default: tests/unit/test_utils.py::test_directory_selinux_type 
    default: PASSED
    default:              [ 77%]
    default: tests/unit/test_utils.py::test_directory_selinux_context 
    default: PASSED
    default:           [ 78%]
    default: tests/unit/test_utils.py::test_directory_acl 
    default: PASSED
    default:                       [ 79%]
    default: tests/unit/test_utils.py::test_user_ownership_int 
    default: PASSED
    default:                  [ 80%]
    default: tests/unit/test_utils.py::test_user_ownership_str 
    default: PASSED
    default:                  [ 81%]
    default: tests/unit/test_utils.py::test_user_ownership_404 
    default: PASSED
    default:                  [ 82%]
    default: tests/unit/test_utils.py::test_group_ownership 
    default: PASSED
    default:                     [ 82%]
    default: tests/unit/test_utils.py::test_group_ownership_str 
    default: PASSED
    default:                 [ 83%]
    default: tests/unit/test_utils.py::test_group_ownership_404 
    default: PASSED
    default:                 [ 84%]
    default: tests/unit/test_utils.py::test_ownership 
    default: PASSED
    default:                           [ 85%]
    default: tests/unit/test_utils.py::test_graceful_get 
    default: PASSED
    default:                        [ 86%]
    default: tests/unit/test_utils.py::test_http_client_get_url 
    default: PASSED
    default:                 [ 87%]
    default: tests/unit/test_utils.py::test_volume_create_from_tuple[/target/path-/target/path] 
    default: PASSED
    default:  [ 88%]
    default: tests/unit/test_utils.py::test_volume_create_from_tuple[input_parameter1-/source/path:/target/path] 
    default: PASSED
    default:  [ 88%]
    default: tests/unit/test_utils.py::test_volume_create_from_tuple[input_parameter2-/source/path:/target/path:mode] 
    default: PASSED
    default:  [ 89%]
    default: tests/unit/test_utils.py::test_volume_create_from_tuple[input_parameter3-/source/path:/target/path] 
    default: PASSED
    default:  [ 90%]
    default: tests/unit/test_utils.py::test_volume_create_from_tuple[input_parameter4-/source/path:/target/path:mode] 
    default: PASSED
    default:  [ 91%]
    default: tests/unit/test_utils.py::test_volume_init[None-/target/path-None-/target/path] 
    default: PASSED
    default:  [ 92%]
    default: tests/unit/test_utils.py::test_volume_init[/source/path-/target/path-None-/source/path:/target/path] 
    default: PASSED
    default:  [ 93%]
    default: tests/unit/test_utils.py::test_volume_init[/source/path-/target/path-mode-/source/path:/target/path:mode] 
    default: PASSED
    default:  [ 94%]
    default: tests/unit/test_utils.py::test_volume_init[source3-/target/path-None-/source/path:/target/path] 
    default: PASSED
    default:  [ 94%]
    default: tests/unit/test_utils.py::test_volume_init[source4-/target/path-mode-/source/path:/target/path:mode] 
    default: PASSED
    default:  [ 95%]
    default: tests/unit/test_utils.py::test_volume_init_raw[instance0-/target/path] 
    default: PASSED
    default:  [ 96%]
    default: tests/unit/test_utils.py::test_volume_init_raw[instance1-/source/path:/target/path] 
    default: PASSED
    default:  [ 97%]
    default: tests/unit/test_utils.py::test_volume_init_raw[instance2-/source/path:/target/path:mode] 
    default: PASSED
    default:  [ 98%]
    default: tests/unit/test_utils.py::test_volume_init_raw[instance3-/source/path:/target/path] 
    default: PASSED
    default:  [ 99%]
    default: tests/unit/test_utils.py::test_volume_init_raw[instance4-/source/path:/target/path:mode] 
    default: PASSED
    default:  [100%]
    default: =================================== FAILURES ===================================
    default: _______________________________ test_probes_port _______________________________
    default:     def test_probes_port():
```

**Comment by jscotka** [comment]
@TomasTomecek or @rpitonak  if you have time, could you please review this PR, I'm not sure who other else is familiar with CONU or nspaw at all :-)
to be able to merge it and make nspawn already working.

File: tests/integration/test_nspawn_backend.py [hunk 1]
```diff
@@ -31,8 +31,7 @@
 
 logger = logging.getLogger("conu.tests")
 image_name = "Fedora-Cloud-Base-27-1.6.x86_64"
-url = "https://download.fedoraproject.org/pub/fedora/linux/development/28/CloudImages/" \
-    "x86_64/images/Fedora-Cloud-Base-28-20180310.n.0.x86_64.raw.xz"
+url = "http://download.eng.brq.redhat.com/pub/fedora/linux/releases/28/Cloud/x86_64/images/Fedora-Cloud-Base-28-1.1.x86_64.raw.xz"
```
[the remark this answers is not in the document]
Comment (jscotka) [reply tests/integration/test_nspawn_backend.py]: ah, thanks, I've missed that it changed

**Comment by jscotka** [comment]
@rpitonak, seems like that this issue is not relevant to nspawn changes.
I also don't understand this but I think I've met also this issue, and it was before Vagrant file fixing. so you used all commits? or just first one?
It were not documented. but there exists issue to do it. So I'll try to do at lease minimal doc in some next PR, to have Nspawn somehow documented.

**Pull request merged**